nxcomp/Socket.cpp

```diff
@@ -743,3 +745,61 @@ int GetHostAddress(const char *name)
     return (*((int *) host -> h_addr_list[0]));
   }
 }
+
+bool IsPort(char* value, long *port)
```

Comment:
Outside of its original class context, the names `IsPort` and `GetUnixSocketPath` are not very informative. `ParseArgAsPortNumber` and `ParseArgAsUnixSocketPath`would be better names.

In any case, is there any reason to not just use a `ChannelEndPoint` object to represent the listening address?